examples/step-67/doc/intro.dox

```diff
@@ -0,0 +1,686 @@
+2 in the reduction of the parameter $\lambda$, since $\|\mathbf{u}\|^2+c^2 \leq
+\|\mathbf{u}\|^2+2 c |\mathbf{u}\| + c^2 = \left(\|\mathbf{u}\|+c\right)^2
+\leq 2 \left(\|\mathbf{u}\|^2+c^2\right)$, with the last inequality following
+from Young's inequality.
+
+The second numerical flux is one proposed by Harten, Lax and van Leer, called
+the HLL flux. It takes the different directions of propagation of the Euler
+equations into account, depending on the speed of sound. It utilizes some
+intermediate states $\bar{\mathbf{u}}$ and $\bar{c}$ to define the two
+branches $s^\mathrm{p} = \max\left(0, \bar{\mathbf{u}}\cdot \mathbf{n} +
+\bar{c}\right)$ and $s^\mathrm{n} = \min\left(0, \bar{\mathbf{u}}\cdot
+\mathbf{n} - \bar{c}\right)$. From these branches, one then defines the flux
+@f[
+\hat{\mathbf{F}}(\mathbf{w}^-,\mathbf{w}^+) =
+\frac{s^\mathrm{p} \mathbf{F}(\mathbf{w}^-)-s^\mathrm{n} \mathbf{F}(\mathbf{w}^+)}
+                   {s^\mathrm p - s^\mathrm{n} } +
+\frac{s^\mathrm{p} s^\mathrm{n}}{s^\mathrm{p}-s^\mathrm{n}}
+\left[\mathbf{w}^--\mathbf{w}^+\right]\otimes \mathbf{n^-}.
+@f]
+Regarding the definition of the intermediate state $\bar{\mathbf{u}}$ and
+$\bar{c}$, several variants have been proposed. The variant originally
+proposed uses a density-averaged definition of the velocity, $\bar{\mathbf{u}}
+= \frac{\sqrt{\rho^-} \mathbf{u}^- + \sqrt{\rho^+}\mathbf{u}^+}{\sqrt{\rho^-}
++ \sqrt{\rho^+}}$. Since we consider the Euler equations without shocks, we
+simply use arithmetic means, $\bar{\mathbf{u}} = \frac{\mathbf{u}^- +
+\mathbf{u}^+}{2}$ and $\bar{c} = \frac{c^- + c^+}{2}$, with $c^{\pm} =
+\sqrt{\gamma p^{\pm} / \rho^{\pm}}$, in this tutorial program, and leave other
+variants to a possible extension. We also note that the HLL flux has been
+extended in literature to the so-called HLLC flux, where C stands for the
+ability to represent so-called contact discontinuities.
+
+At the boundaries with no neighboring state $\mathbf{w}^+$ available, it is
+common practice to deduce suitable exterior values from the boundary
+conditions (see the general literature on DG methods for details). In this
+tutorial program, we consider four types of boundary conditions, namely
+<b>inflow boundary conditions</b> where all components are prescribed,
+@f[
+\mathbf{w}^+ = \begin{pmatrix} \rho_\mathrm{D}(t)\\
+(\rho \mathbf u)_{\mathrm D}(t) \\ E_\mathrm{D}(t)\end{pmatrix} \quad
+ \text{(Dirichlet)},
+@f]
+<b>subsonic outflow boundaries</b>, where we do not prescribe exterior
+solutions as the flow field is leaving the domain and use the interior values
+instead; we still need to prescribe the energy as there is one incoming
+characteristic left in the Euler flux,
+@f[
+\mathbf{w}^+ = \begin{pmatrix} \rho^-\\
+(\rho \mathbf u)^- \\ E_\mathrm{D}(t)\end{pmatrix} \quad
+ \text{(mixed Neumann/Dirichlet)},
+@f]
+<b>supersonic outflow boundaries</b>, where all characteristics are outgoing
```

Comment:
That's a question: I put it in because it is easy to implement, on the other hand all examples with supersonic flows I tried were having too many oscillations and failed due to negative density. If nobody objects, I will remove it and put the code to the possibilities for extensions.

Reply:
This sounds like a good idea!